Scripts/setup.sh

```
@@ -103,16 +108,23 @@ if ! command -v swiftformat &> /dev/null; then
     fi
 fi
 
-# Check Ruby version (require Ruby 3.3.3)
+# Check Ruby version (require Ruby 3.3.0 or higher)
 RUBY_VERSION=$(ruby -v | awk '{print $2}' | cut -d'p' -f1)
-if [ "$RUBY_VERSION" != "3.3.3" ]; then
+RUBY_MAJOR=$(echo "$RUBY_VERSION" | cut -d'.' -f1)
+RUBY_MINOR=$(echo "$RUBY_VERSION" | cut -d'.' -f2)
+RUBY_PATCH=$(echo "$RUBY_VERSION" | cut -d'.' -f3)
```

Shellcheck (0.10.0)

[warning] RUBY_PATCH appears unused. Verify use (or export if used externally).

(SC2034)